edxval/api.py

```diff
@@ -296,11 +341,41 @@ def get_video_info_for_course_and_profile(course_id, profile_name):
         raise ValInternalError(error_message)
 
     # DRF serializers were causing extra queries for some reason...
-    return {
-        enc_vid.video.edx_video_id: {
-            "url": enc_vid.url,
-            "file_size": enc_vid.file_size,
-            "duration": enc_vid.video.duration,
-        }
-        for enc_vid in encoded_videos
-    }
+    return_dict = {}
+    for enc_vid in encoded_videos:
+        return_dict.setdefault(enc_vid.video.edx_video_id, {}).update(
+            {
+                enc_vid.profile.profile_name: {
+                    "url": enc_vid.url,
+                    "file_size": enc_vid.file_size,
+                    "duration": enc_vid.video.duration,
```

Comment:
It's a little weird to include the duration with each encoding, when it's a property of the video and not the encoding. The fact that the sole intended caller of this ends up filtering out the duration suggests that we should consider modifying the returned object to something like the following:

```
{
    "video_id_1": {
        "duration": 100,
        "encodings": {
            "profile1": {
                "url": "http://example.com/foo.mp4",
                "file_size": 555,
            },
            ...
        }
    },
    ...
```